app/Jobs/IndexTransactions.php

```diff
@@ -9,6 +9,13 @@
 
 final class IndexTransactions extends IndexModel
 {
+    public function uniqueId(): string
+    {
+        $latestIndexedTimestamp = $this->getLatestIndexedTimestamp('transactions');
+
+        return  'transactions:'.$latestIndexedTimestamp;
```

Comment:
Wouldn't these unique ids be better being just the index name? :thinking: scenario where they somehow overlap:

- job starts with `transactions:1` and pulls 10 transactions
- 2nd job starts while first job is running, but the latest timestamp hasn't been updated - resulting in the same 10 (or more) transactions being indexed again